services/symfony/symfony-insight.service.js

```diff
@@ -0,0 +1,277 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseXmlService = require('../base-xml')
+const serverSecrets = require('../../lib/server-secrets')
+
+const violationSchema = Joi.object({
+  severity: Joi.string()
+    .regex(/info|minor|major|critical/)
+    .required(),
+}).required()
+
+const schema = Joi.object({
+  project: Joi.object({
+    'last-analysis': Joi.object({
+      status: Joi.string()
+        .regex(/ordered|running|measured|analyzed|finished/)
+        .required(),
+      grade: Joi.string().regex(/platinum|gold|silver|bronze|none/),
+      violations: Joi.object({
+        // RE: https://github.com/NaturalIntelligence/fast-xml-parser/issues/68
+        // The BaseXmlService uses the fast-xml-parser which doesn't support forcing
+        // the xml nodes to always be parsed as an array. Currently, if the response
+        // only contains a single violation then it will be parsed as an object,
+        // otherwise it will be parsed as an array.
```

Comment:
Does [`Joi.array().single()`](https://github.com/hapijs/joi/blob/master/API.md#arraysingleenabled) help with this?

Reply:
It looks like it does!